app.rb

```diff
@@ -79,12 +79,18 @@ def config
         id_token = token_response[:id_token]
         userinfo_response = userinfo(id_token)
 
-        session[:login_msg] = 'ok'
-        session[:logout_uri] = logout_uri(token_response[:id_token])
-        session[:userinfo] = userinfo_response
-        session[:email] = session[:userinfo][:email]
+        if session.delete(:step_up_enabled)
+          aal = session.delete(:step_up_aal)
 
-        redirect to('/')
+          redirect to("/auth/request?aal=#{aal}&ial=2")
```

Comment:
Do we validate `aal` in any way here? This has a light odor of a semi-open redirect.

Reply:
Done in e65cbd6